Expand mutual TLS security considerations (#1786)
moq-wg/moq-transport: draft-ietf-moq-transport.md

## Authorization {#sec-authorization}

MOQT supports authorization via mutual TLS (mTLS) for endpoint
identification and via token-based schemes for fine-grained,
application-defined access control. The two mechanisms can be used together.

### Mutual TLS {#sec-mtls}

In mutual TLS, both peers present an X.509 certificate during the TLS 1.3
handshake ({{?RFC8446}}), carried in the underlying transport. An endpoint that
verifies a server certificate does so following {{?RFC9525}}.
An application that authenticates clients via mTLS defines how a client
certificate maps to identity.

> 
> Once a peer is authenticated, an application MAY use
> attributes in the peer's certificate as an input
> to authorization decisions; the granularity and policy of such authorization
> is out of scope for this document.
> 

Author: ```suggestion
In mutual TLS, both peers present an X.509 certificate during the TLS 1.3
handshake ({{?RFC8446}}), carried in the underlying transport. An endpoint that
verifies a server certificate does so following {{?RFC9525}}.  An application
that authenticates clients via mTLS defines how a client certificate maps to
identity.

Once a peer is authenticated, an application MAY use attributes in the peer's
certificate as an input to authorization decisions; the granularity and policy
of such authorization is out of scope for this document.
```

## Authorization {#sec-authorization}

MOQT supports authorization via mutual TLS (mTLS) for endpoint
identification and via token-based schemes for fine-grained,
application-defined access control. The two mechanisms can be used together.

### Mutual TLS {#sec-mtls}

In mutual TLS, both peers present an X.509 certificate during the TLS 1.3
handshake ({{?RFC8446}}), carried in the underlying transport. An endpoint that
verifies a server certificate does so following {{?RFC9525}};

> an application that authenticates clients via mTLS defines how a client

Reviewer: ```suggestion
verifies a server certificate does so following {{?RFC9525}}.
An application that authenticates clients via mTLS defines how a client
```

certificate maps to identity.

mTLS can be used between relays and between a relay and an end subscriber or

> original publisher. Once a peer is authenticated, an application MAY use the

Author: ```suggestion
Once a peer is authenticated, an application MAY use
```

authenticated identity, or attributes in the peer's certificate, as an input
to authorization decisions; the granularity and policy of such authorization

> is out of scope for this document.

Author: ```suggestion
attributes in the peer's certificate as an input
to authorization decisions; the granularity and policy of such authorization
is out of scope for this document.
```

Two properties of mTLS are significant for MOQT:

- TLS authentication is hop-by-hop: it authenticates only the immediate
  transport peer, not endpoints reached through intervening relays.
  Authorization that needs to survive relaying is better carried in tokens
  (see {{sec-tokens}}).

- Client authentication occurs at connection establishment. Neither native
  QUIC nor WebTransport can request a client certificate after the handshake,
  so all requests on a session share the identity established at connection

>   time.

Author: ```suggestion
```

## Authorization {#sec-authorization}

MOQT supports authorization via mutual TLS (mTLS) for endpoint
identification and via token-based schemes for fine-grained,
application-defined access control. The two mechanisms can be used together.

### Mutual TLS {#sec-mtls}

In mutual TLS, both peers present an X.509 certificate during the TLS 1.3
handshake ({{?RFC8446}}), carried in the underlying transport (native QUIC

> {{?RFC9001}} or the HTTP connection underlying WebTransport). An endpoint that

Author: ```suggestion
handshake ({{?RFC8446}}), carried in the underlying transport. An endpoint that
```

> verifies a server certificate's identity does so following {{?RFC9525}};

Author: ```suggestion
verifies a server certificate does so following {{?RFC9525}};
```

{{?RFC9525}} does not define verification of client certificate identities, so
an application that authenticates clients via mTLS defines how a client

> certificate is validated and matched against the identities it trusts.

Author: ```suggestion
an application that authenticates clients via mTLS defines how a client
certificate maps to identity.
```

## Authorization {#sec-authorization}

MOQT supports authorization via mutual TLS (mTLS) for endpoint
identification and via token-based schemes for fine-grained,
application-defined access control. The two mechanisms can be used together.

### Mutual TLS {#sec-mtls}

In mutual TLS, both peers present an X.509 certificate during the TLS 1.3
handshake ({{?RFC8446}}) carried within the underlying transport: directly
for native QUIC ({{?RFC9001}}), or within the HTTP connection underlying
WebTransport. An endpoint authenticates its peer by validating the presented
certificate chain to a configured trust anchor, following the certificate

> path validation rules of {{?RFC5280}}.

Reviewer 2: I will note that this is very often not what happens in practice (for client certs, though you can bypass that for server certs too), so we should omit this sentence or says that application does "something".

An endpoint that verifies a server certificate's identity does so following
the procedures in {{?RFC9525}}. {{?RFC9525}} does not define verification of

> client certificate identities; a deployment that authenticates clients via

Reviewer 2: Do we use term "deployment" elsewhere, or should we stick with something like "application"?
